zbus_xmlgen/src/main.rs

```diff
@@ -113,29 +114,55 @@ fn main() -> Result<(), Box<dyn Error>> {
         }
     };
 
+    let split_interfaces = args().any(|arg| arg == "--split-interfaces");
+
     let fdo_iface_prefix = "org.freedesktop.DBus";
     let (fdo_standard_ifaces, needed_ifaces): (Vec<&Interface<'_>>, Vec<&Interface<'_>>) = node
         .interfaces()
         .iter()
         .partition(|&i| i.name().starts_with(fdo_iface_prefix));
 
-    let output = write_interfaces(
-        &needed_ifaces,
-        &fdo_standard_ifaces,
-        service,
-        path,
-        &input_src,
-    )?;
-    println!("{}", output);
+    if !split_interfaces {
+        let output = write_interfaces(
+            &needed_ifaces,
+            &fdo_standard_ifaces,
+            &service,
+            &path,
+            &input_src,
+        )?;
+        println!("{}", output);
+        return Ok(());
+    }
+
+    for interface in needed_ifaces {
+        let output = write_interfaces(
+            &[&interface],
+            &fdo_standard_ifaces,
+            &service,
+            &path,
+            &input_src,
+        )?;
+        std::fs::write(
+            format!(
+                "{}.rs",
+                interface
+                    .name()
+                    .split('.')
+                    .last()
+                    .expect("Failed to split name")
+            ),
+            output,
+        )?;
+    }
 
     Ok(())
 }
 
 fn write_interfaces(
     interfaces: &[&Interface<'_>],
     standard_interfaces: &[&Interface<'_>],
-    service: Option<BusName<'_>>,
-    path: Option<ObjectPath<'_>>,
+    service: &Option<BusName<'_>>,
+    path: &Option<ObjectPath<'_>>,
```

Comment:
Never seen Option being passed as ref and seems unrelated to the commit. The name structs are free (no runtime cost) to clone so I'd not bother.